instrumentation/oshi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/oshi/MetricsRegistration.java

```diff
@@ -17,13 +19,28 @@ public final class MetricsRegistration {
 
   public static void register() {
```

Comment:
Did you consider returning `List<AutoCloseable>` or perhaps just `AutoCloseable` from here and letting user decide how and when they want to shut it down?

Reply:
Thank you for the review @laurit.

I did not do it here but did it for the [library registration](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10364/files#diff-4797d9a38ce25a614ac77dc83b96b5aae5df9221455022fe953f6ab234407f78R25) which now returns the `List<AutoCloseable>`. 

This javaagent is a user of that library, so did not change for javaagent. The caller is [SystemInfoInstrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/8bdf6148097dd2d4a0bbc8c3c799da0c9d609a86/instrumentation/oshi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/oshi/SystemInfoInstrumentation.java#L40) which is generally just delegating registrations (as seen in some other instrumentations as well).

```
  public static class GetCurrentPlatformEnumAdvice {

    @Advice.OnMethodEnter(suppress = Throwable.class)
    public static void onEnter() {
      MetricsRegistration.register();
    }
  }
``` 

On the other hand, the library can be used independently and outside of this instrumentation. That user can decide how to handle the closure.

Reply:
Sorry, my bad. Though that this was for library instrumentation.

Reply:
no worries, thanks for checking.